src/WooCommerce.php

```diff
@@ -1513,4 +1514,16 @@ private static function getRepricingOptionsDropdownValues(\WC_Product $product)
     return $dropdown_values;
   }
 
+   /**
+   * Invalidates the object cache in Woocommerce order status manager.
+   * 
+   * @implements save_post_wc_order_status
+   * 
+   */
+  public static function save_post_wc_order_status() {
+    if (wp_cache_get('wc_order_status_manager_order_status_posts')) {
```

Comment:
1. Is there a wp_cache_has() that only looks up but does not fetch the cache item? If not, please use the `$found` argument of wp_cache_get() to determine whether the cache item was found.

2. This fix is so trivial, why aren't we contributing it as a patch to the plugin vendor (upstream PR if possible?) and temporarily apply it via shared-patches to the actual plugin instead?  Otherwise we'll have to remove this code fairly soon again once upstream is fixed?

Reply:
Hi, @sun, Thank you for your great review, 
I searched and the method wp_cache_has() does not exist, and also can you explain what you mean:
**use the $found argument of wp_cache_get() to determine whether the cache item was found.**

Reply:
`if (false === wp_cache_get('wc_order_status_manager_order_status_posts'))`

Reply:
Just recognized that this is indeed not noted in the manual yet, so I explained it here: https://developer.wordpress.org/reference/functions/wp_cache_get/#comment-6157

The post is pending moderation approval, so here's a copy:

---
Here is how you use the `$found` parameter in your code – it is passed by reference, so you do not need to define the variable upfront, its value will be set by the called function instead:

```php
$value = wp_cache_get('mykey', 'mygroup', false, $found);
if (!$found) {
  $value = my_expensive_function();
  wp_cache_set('mykey', $value, 'mygroup');
}
return $value;
```
You need to use the `$found` parameter if there is a slightest chance that the value you are caching can be zero, false, null, or otherwise equivalent to false like an empty array.

If you are not sure, or if you are looking for **a general rule of thumb – always use the `$found` parameter.**

If some object caches are not implementing the `$found` parameter correctly, then that is a good reason to fix those implementations (or use better maintained ones) and not a reason to not use the parameter.

Reply:
Did you work with https://github.com/makers99/wp-cli-shared-patches already?

Reply:
> 

Thank you so much for your awesome clarification 👍
Unfortunately, I didn't work with  https://github.com/makers99/wp-cli-shared-patches yet.